ci/steps/attach_logs.py

```diff
@@ -1,179 +1,243 @@
+    flavour_set: str,
     gardenlinux_epoch: str,
-    modifiers: str,
-    platform: str,
+    is_package_build: bool,
+    manifest_set_key: str,
+    platform_set: str,
+    promote_target: str,
     repo_dir: str,
     version: str,
-):
+) -> bool:
     cicd_cfg = glci.util.cicd_cfg(cfg_name=cicd_cfg_name)
     s3_client = glci.s3.s3_client(cicd_cfg)
     aws_cfg_name = cicd_cfg.build.aws_cfg_name
-    s3_bucket_name = cicd_cfg.build.s3_bucket_name
-    s3_package_bucket_name = cicd_cfg.package_build.s3_bucket_name
 
     log_path = os.path.join(repo_dir, 'build_log_full.zip')
     if not os.path.exists(log_path):
         print("No file found with log files, won't upload.")
         print("Exiting with failure, see logs from previous steps")
         return False
 
-    build_target_set = glci.model.BuildTarget.set_from_str(build_targets)
-    if not glci.model.BuildTarget.MANIFEST in build_target_set:
-        print(f'build target {glci.model.BuildTarget.MANIFEST=} not specified - skip upload')
+    prefix = datetime.datetime.utcnow().strftime('%Y%m%d-%H%M%S') + '-' + \
+        committish[:6] + '-'
+
+    if is_package_build:
+        s3_key = 'logs/' + prefix + 'package_build_log.zip'
+        s3_bucket_name = cicd_cfg.package_build.s3_bucket_name
+    else:
+        s3_key = 'objects/' + prefix + 'build_log.zip'
+        s3_bucket_name = cicd_cfg.build.s3_bucket_name
+
+    print(f'uploaded zipped logs to {s3_key=}')
+    uploaded_file = _upload_file(
+        log_file_path=log_path,
+        s3_key=s3_key,
+        s3_client=s3_client,
+        s3_bucket_name=s3_bucket_name
+    )
+
+    # write a file with the download URL so that it can be later added to the email
+    with open(os.path.join(repo_dir, 'log_url.txt'), 'w') as f:
+        f.write(f'https://gardenlinux.s3.eu-central-1.amazonaws.com/{s3_key}')
+
+    if is_package_build:
         return True
 
-    manifest_available = True
-    if not platform.strip() or not modifiers.strip():
-        print('No platform or modifiers given, cannot find release manifest, log upload skipped.')
-        print('This is probably a package build.')
-        manifest_available = False
-
-    if manifest_available:
-        print(f'downloading release manifest from s3 {aws_cfg_name=} {s3_bucket_name=}')
-        find_release = glci.util.preconfigured(
-            func=glci.util.find_release,
-            cicd_cfg=glci.util.cicd_cfg(cicd_cfg_name)
+    # if we create manifests attach logs to all manifests that actually have been build in this run
+    build_target_set = glci.model.BuildTarget.set_from_str(build_targets)
+    if glci.model.BuildTarget.MANIFEST in build_target_set:
+        _attach_logs_to_single_manifest(
+            build_dict_str=build_dict_str,
+            s3_client=s3_client,
+            s3_bucket_name=s3_bucket_name,
+            s3_log_key=s3_key,
         )
+    else:
+        print(f'build target {glci.model.BuildTarget.MANIFEST=} not specified - do not attach logs')
```

Comment:
nit: maybe rather state `will not attach logs` - otherwise, this reads to me like an instruction to the reader

Reply:
Changed.